ruby/hangman/game.rb

```diff
@@ -0,0 +1,189 @@
+# THIS IS BASICALLY HANGMAN.
+
+# CLASS PSEUDOCODE:
+
+# Class vars: gameboard, original word, wrong guesses
+
+# CREATE THE GAMEBOARD
+# 1. takes: The user input (string)
+# 2. gets the length of that input
+# 2.5 gets the length of the input without spaces
+# 3. Returns a string the same length of the user input with
+# 	non letter characters still there, everything else
+# 	replaced with _ - this is the GAMEBOARD
+
+# PROCESS THE GUESS
+# 1. Takes in the letter
+# 2. If longer than 1 char, returns error
+# 3. If exists in original word, update GAMEBOARD
+# 	RETURN TRUE
+# 4. if not, increment wrong guesses by one
+# 	RETURN FALSE
+
+
+# UPDATE GAMEBOARD
+# 1. Takes the letter
+# 2. Replaces all instances of _ with the letter.
+# 3. Updates the GAMEBOARD
+# 4. Returns whatever it wants, that doesn't matter.
+
+# FINAL MESSAGE
+# 1. takes nothing
+# 2. if GAMEBOARD doesn't have _s then print something nice
+# 3. If GAMEBOARD does have _s then print something rude
+
+class Game
+=begin
+	@original_word
+	@gameboard
+	@guesses_remaining
+	@prev_guesses 
+	@win
+=end
+	attr_reader :gameboard, :guesses_remaining, :prev_guesses, :win
+
+	def initialize(word)
+		@original_word = word.downcase
+		@gameboard = ""
+		@prev_guesses = []
+		@win = false
+
+		letter_replace = "_ "
+
+		words = word.split("")
+		words.each do |letter|
+			if letter.downcase.match(/^[[:alpha:]]+$/) #Tbh, I'm not sure if this is easier than pasting in the alphabet string
+				@gameboard << letter_replace
+			elsif " _".include? letter
+				@gameboard << "  "
+			else
+				@gameboard.chomp!(" ") << letter 
+			end
+		end
+
+		establish_guess_limit
+		@gameboard.chomp!(" ")	
+	end
+	
+	#Processes a user's guess. Assumes a single alpha char input.
+	def process_guess(letter)
+
+		letter = letter.downcase
+
+		if @prev_guesses.include? letter
+			puts "\"#{letter}\" has already been guessed"
+			print_status
+			return false 
+		else
+			@prev_guesses << letter
+			decrement_guesses
+		end
+
+		if @original_word.include? letter
+			update_gameboard(letter)
+			print_status
+			true
+		else
+			print_status
+			false
+		end
+	end
+
+	def update_gameboard(letter)
+		
+		special_char_index_modifier = 0
+
+		@gameboard.split("").each_index do |position|
+			if !(@gameboard[position].match(/^[[:alpha:]]+$/)) && !(" _".include? @gameboard[position])
+				special_char_index_modifier += 1
+			end
+
+			if (@gameboard[position] == "_") && (@original_word[(position / 2) + special_char_index_modifier] == letter)
+				@gameboard[position] = letter
+			end
+		end
+		@gameboard
+	end
+
+	def is_over?
+		if !(@gameboard.include? "_")
+			@win = true
+		elsif guesses_remaining == 0 
+			true
+		else
+			false
+		end
+	end
+
+	def final_message
+		if @win
```

Comment:
Instead of checking the `@win` variable, you have a method that can return true or false based on your win conditions. `if is_over?` would make sense here and would remove the need for an `@win` variable.